server/models/Profile.js

```diff
@@ -25,15 +25,21 @@ const profileSchema = new mongoose.Schema({
   },
   dateOfBirth: {
     type: Date,
-    default: null
+    default: null,
   },
   availability: {
     type: [Date],
-    default: []
+    default: [],
   },
   photo: {
-    type: String,
-    default: "",
+    url: {
+      type: String,
+      default: "",
+    },
+    key: {
```

Comment:
Optional: wouldn't it be better to use the key from the S3 URL as opposed to storing it and retrieving it from the Profile model  :)

Reply:
I'll refactor this once I've finalized how to use the s3 bucket. Right now I'm generating a signed url so that images from the private s3 bucket can be publicly viewed. As of now, generating a signed url is the best solution I have without exposing the s3 bucket to public access.

Reply:
I will refactor this once I've finalized how I will use s3. Initially I was using the default url generated by s3, but that url is not publicly viewable for a private s3 bucket. Now I'm generating a pre-signed url, which is publicly viewable for a private s3 bucket. However, the downside is that the pre-signed url is only valid up to a week. @rajivtitus if you have a suggestion for how to keep the s3 bucket access private while generating a public url that is valid for more than a week, let me know. As far as I can tell, I can only use signature version 4 because of when the bucket was created. The [aws documentation](https://aws.amazon.com/premiumsupport/knowledge-center/presigned-url-s3-bucket-expiration/) provides a few solutions to achieve what I want, but I'm not sure which one would be best. Any thoughts?

Reply:
Hey Ankit! After reviewing the docs, I would go with 'Amazon S3: Allows Amazon Cognito users to access objects in their bucket' option because it provides authentication, authorization, and user management even through third party services. However, you can make the S3 bucket public to have indefinite access to the URL (to make things easier for our project)

Reply:
Thanks for the recommendation! I'll try out the Amazon Cognito users option first, if I can't get it to work. I'll make the bucket public.